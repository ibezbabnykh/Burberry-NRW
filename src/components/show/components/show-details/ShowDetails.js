import React from 'react';
import PropTypes from 'prop-types';
import ReactHtmlParser from 'react-html-parser';

import './ShowDetails.scss';

const ShowDetails = ({ data }) => {
  const { name, image, summary } = data;

  return (
    <article className="show-details">
      <h1 className="show-details_title">{name}</h1>
      <div className="show-details_content">
        <div className="show-details_img">
          <a href={image.original} rel="noreferrer" target="_blank">
            <img src={image.medium} alt={name} title={name} />
          </a>
        </div>
        <div className="show-details_description">{ReactHtmlParser(summary)}</div>
      </div>
    </article>
  );
}

ShowDetails.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    image: PropTypes.shape({
      original: PropTypes.string,
      medium: PropTypes.string,
    }).isRequired,
  }),
};

export default ShowDetails;