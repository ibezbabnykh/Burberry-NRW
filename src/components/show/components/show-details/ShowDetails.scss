@import '../../../../assets/style/variables.scss';

.show-details {
  margin-bottom: 30px;
  
  &_content {
    display: flex;
    flex-flow: column;
  
    @media (min-width: $screen-sm) {
      flex-flow: row;
    }
  }

  &_title {
    margin: 0 0 20px;
  }

  &_img {
    margin-bottom: 20px;
    text-align: center;

    @media (min-width: $screen-sm) {
      margin-right: 20px;
    }

    img {
      vertical-align: top;
    }
  }

  &_description {
    flex-grow: 1;

    p {
      margin: 0 0 15px;
    }
  }
}