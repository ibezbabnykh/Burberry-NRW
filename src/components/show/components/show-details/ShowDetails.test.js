import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ShowDetails from './ShowDetails';

configure({ adapter: new Adapter() });

const mockProps = {
  data: {
    image: {
      medium: 'https://via.placeholder.com/250x140',
      original: 'https://via.placeholder.com/1280x720'
    },
    name: 'show name',
    summary: 'show summary'
  }
};

describe('ShowDetails component', () => {
  it('should render', () => {
    const component = toJson(shallow(<ShowDetails {...mockProps} />));
    expect(component).toMatchSnapshot();
  });
});