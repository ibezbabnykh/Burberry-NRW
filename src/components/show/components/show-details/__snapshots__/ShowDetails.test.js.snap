// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowDetails component should render 1`] = `
<article
  className="show-details"
>
  <h1
    className="show-details_title"
  >
    show name
  </h1>
  <div
    className="show-details_content"
  >
    <div
      className="show-details_img"
    >
      <a
        href="https://via.placeholder.com/1280x720"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="show name"
          src="https://via.placeholder.com/250x140"
          title="show name"
        />
      </a>
    </div>
    <div
      className="show-details_description"
    >
      show summary
    </div>
  </div>
</article>
`;
